compiler/testData/codegen/box/strings/stringFromJavaPlus.kt

```diff
@@ -0,0 +1,138 @@
+// IGNORE_BACKEND_FIR: JVM_IR
+// TARGET_BACKEND: JVM
+// WITH_RUNTIME
+
+// FILE: J.java
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class J {
+    public static String platformStringIsNull() {
+        return null;
+    }
+
+    @NotNull
+    public static String notNullStringIsNull() {
+        return null;
+    }
+
+    @Nullable
+    public static String nullableStringIsNull() {
+        return null;
+    }
+
+    @NotNull
+    public static String notNullStringIsNotNull() {
+        return "foo";
+    }
+
+    public static String platformStringIsNotNull() {
+        return "foo";
+    }
+
+    @Nullable
+    public static String nullableStringIsNotNull() {
+        return "foo";
+    }
+}
+
+// FILE: main.kt
+import kotlin.test.assertEquals
+
+fun box(): String {
+    val n = 123
+
+    // Due to KT-36625, certain concatenation calls below (marked with a comment) SHOULD fail a nullability check but do not.
```

Comment:
Minor: not clear if it's really "SHOULD". Language design formalities are not settled in this case.
I'd change this comment to something like 
```
// Null check behavior in string concatenation might change depending on language design decision for KT-36625.
```